osu.Game/Rulesets/Objects/Pooling/PoolableDrawableWithLifetime.cs

```diff
@@ -117,11 +121,23 @@ private void free()
 
             OnFree(Entry);
 
+            Entry.LifetimeChanged -= entryLifetimeChanged;
             Entry = null;
-            base.LifetimeStart = double.MinValue;
-            base.LifetimeEnd = double.MaxValue;
+            setLifetimeFromEntry();
 
             HasEntryApplied = false;
         }
+
+        private void entryLifetimeChanged(LifetimeEntry entry)
+        {
+            Debug.Assert(entry == Entry);
+            setLifetimeFromEntry();
+        }
+
+        private void setLifetimeFromEntry()
+        {
+            base.LifetimeStart = Entry?.LifetimeStart ?? double.MinValue;
+            base.LifetimeEnd = Entry?.LifetimeEnd ?? double.MaxValue;
+        }
```

Comment:
This is also a bit baffling to me - a `setLifetimeFromEntry()` method that is not only null-tolerant wrt the entry, but also _does things_ when there is no entry. I'd just inline the `{Min,Max}Value` restoration in `free()` if anything, and keep this code path as one that actually needs an entry to work.